### ISSUE TITLE
Correct socket.io server close "sails.lower"

### ISSUE THREAD

**Issue by cheplv** [balderdashy/sails#3081]
Currently https://github.com/balderdashy/sails/blob/master/lib/app/lower.js#L74 uses sails.io.server.close(), instead of sails.io.close() - https://github.com/Automattic/socket.io/blob/master/lib/index.js#L348

Currently code produces exception which is suppressed by "try ... catch" block.
This error prevents calling of "config.sockets.afterDisconnect"

And need to detect if http server (sails.hooks.http) is already closed.

**Comment by CWyrtzen** [comment]
forgive my ignorance, is this confirmed as a bug? just trying to help issues along and prioritize - let me know!

**Comment by cheplv** [comment]
Another one thing - when use connect-mongo as session storage - connections not closed, i.e. need to implement something like "uninitialize" in hooks to free used resources

**Comment by sailsbot** [comment]
Thanks for posting, @cheplv.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
